docs/architecture/adr-023-protobuf-naming.md

```diff
@@ -99,13 +99,67 @@ the best choices for the future.
-break smart contracts/persistent scripts and generally provide a bad UX for
-clients. With protobuf, there should usually be ways to extend existing
-functionality instead of just breaking it.
+#### Guidelines on Stable Package Versions
+
+In general, schema evolution is the way to update protobuf schemas. That means that new fields,
+messages, and RPC methods are _added_ to existing schemas and old fields, messages and RPC methods
+are maintained as long as possible.
+
+Breaking things is often unacceptable in a blockchain scenario. For instance, immutable smart contracts
+may depend on certain data schemas on the host chain. If the host chain breaks those schemas, the smart
+contract may be irreparably broken. Even when things can be fixed (for instance in client software),
+this often comes at a high cost.
+
+Instead of breaking things, we should make every effort to evolve schemas rather than just breaking them.
+[Buf](https://buf.build) breaking change detection should be used on all stable (non-alpha or beta) packages
+to prevent such breakage.
+
+With that in mind, different stable versions (i.e. `v1` or `v2`) of a package should more or less be considered
+different packages and this should be last resort approach for upgrading protobuf schemas. Scenarios where creating
+a `v2` may make sense are:
+* we want to create a new module with similar functionality to an existing module and adding `v2` is the most natural
+way to do this. In that case, there are really just two different, but similar modules with different APIs.
+* we want to add a new revamped API for an existing module and it's just too cumbersome to add it to the existing package,
+so putting it in `v2` is cleaner for users. In this case, care should be made to not deprecate support for
+`v1` if it is actively used in immutable smart contracts.
+
+An alternative to creating `v2` packages is creating `v2` messages and RPC methods. For instance say we found a better API
+for sending coins, then we could add `MsgSendV2` to the existing bank package without needing to change `MsgSend`
+or create a `bank.v2` package.
+
+#### Guidelines on unstable (alpha and beta) package versions
+
+The following guidelines are recommended for marking packages as alpha or beta:
+* marking something as `alpha` or `beta` should be a last resort and just putting something in the
+stable package (i.e. `v1` or `v2`) should be preferred
+* a package *should* be marked as `alpha` *if and only if* there are active discussions to remove
+or significantly alter the package in the near future
+* a package *should* be marked as `beta` *if and only if* there is an active discussion to 
+significantly refactor/rework the functionality in the near future but not remove it
+* modules *can and should* have types in both stable (i.e. `v1` or `v2`) and unstable (`alpha` or `beta`) packages.
+Experimental functionality to existing stable packages could be added as `alpha` for instance.
```

Comment:
Does this mean that we suffix Alpha/Alpha1 in experimental types in a stable package?
e.g. `type MsgSendAlpha1`

Reply:
No... stable packages are break checked so that wouldn't work

Reply:
Ah, no, I meant adding a new type `MsgSendAlpha1` in a stable package. Surely this does not break stability.

My question is:
- are we allowed to add a new type `MsgSendAlpha1` in a stable package?
- if yes, are we allowed to remove it later on? (that would break compat)

Reply:
No we can't remove types from stable packages. Buf only allows exclusions on a per package basis. I think that's sensible too. Let's just add alpha/beta types to alpha/beta packages.

Reply:
> Let's just add alpha/beta types to alpha/beta packages.

Yeah, agreed with that.

```suggestion
```